088-rebalance-progress-status

```diff
@@ -0,0 +1,328 @@
+	excludedBrokersForReplicaMove: []
+	excludedTopics: []
+	intraBrokerDataToMoveMB: 0
+	monitoredPartitionsPercentage: 100
+	numIntraBrokerReplicaMovements: 0
+	numLeaderMovements: 16
+	numReplicaMovements: 0
+	onDemandBalancednessScoreAfter: 95.4347095948149
+	onDemandBalancednessScoreBefore: 89.4347095948149
+	provisionRecommendation: ""
+	provisionStatus: RIGHT_SIZED
+	recentWindows: 1
+  progress:
+        estimatedTimeToCompletion: 5m [2]
+        percentageComplete: 80% [3]
+        rebalanceProgressConfigMap: my-rebalance-progress [4]
+```
+[1] The “progress” section will be visible during the KafkaRebalance “Rebalancing” and  “Stopped” states.
+[2] The minimum estimated time it will take the rebalance to complete.
+[3] The percentage complete of the ongoing rebalance in the range [0-100]%
+[4] The ConfigMap where “non-verbose” JSON payload from Executor State from CruiseControlState endpoint is stored.
+
+### Executor State
+
+All the information needed for the `progress` section proposed above relies on the [ExecutorState](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control) of the CruiseControlState endpoint.
+
+Querying the Executor State during an interbroker balance dumps the following JSON payload:
+```
+{
+  "abortingPartitions": 0,
+  "averageConcurrentPartitionMovementsPerBroker": 5,
+  "finishedDataMovement": 0,
+  "maximumConcurrentPartitionMovementsPerBroker": 5,
+  "minimumConcurrentPartitionMovementsPerBroker": 5,
+  "numFinishedPartitionMovements": 0,
+  "numInProgressPartitionMovements": 0,
+  "numPendingPartitionMovements": 20,
+  "numTotalPartitionMovements": 20,
+  "state": "INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS",
+  "totalDataToMove": 0,
+  "triggeredSelfHealingTaskId": "",
+  "triggeredTaskReason": "No reason provided (Client: 172.17.0.1, Date: 2024-11-15T19:41:27Z)",
+  "triggeredUserTaskId": "0230d401-6a36-430e-9858-fac8f2edde93"
+}
+```
+
+For determining which fields are included for different executor states e.g. NO_TASK_IN_PROGRESS, STARTING_EXECUTION, INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS whether the verbose parameter is provided or not, refer to the code [here](https://github.com/linkedin/cruise-control/blob/2.5.141/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorState.java#L427)
+
+For an exhaustive list of all the fields that could be included, refer to Cruise Control’s OpenAPI spec, refer to the file [here](https://github.com/linkedin/cruise-control/blob/2.5.141/cruise-control/src/main/resources/yaml/responses/executorState.yaml)
+
+The “non-verbose” JSON payload from the ExecutorState is already too verbose to include in the `KafkaRebalance` status in its entirety. 
+However, having the information available to users is still useful especially when debugging the state of a partition rebalance. 
+Therefore, we will store the JSON payload in its own ConfigMap, “rebalanceProgressConfigMap”. 
```

Comment:
```suggestion
Therefore, we will store the JSON payload in its own `ConfigMap`, the name of which will be stored in the `rebalanceProgressConfigMap` field. 
```

Reply:
We should also define the name of such ConfigMap. It's not configurable and user cannot decided this name. I think it will be something pre-formatted starting from the `KafkaRebalance` name?

Reply:
Added a sentence how the name of the `ConfigMap` is generated